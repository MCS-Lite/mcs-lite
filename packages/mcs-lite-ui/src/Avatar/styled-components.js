// @flow
import * as React from 'react';
import styled from 'styled-components';

const Wrapper: React.ComponentType<*> = styled.div`
  img,
  svg {
    display: inline-block;
    border-radius: 50%;
    border: 1px solid ${props => props.theme.color.grayBase};
    padding: 1px;
    background: ${props => props.theme.color.white};
    box-sizing: border-box;
    object-fit: cover;
  }
`;

export default Wrapper;
