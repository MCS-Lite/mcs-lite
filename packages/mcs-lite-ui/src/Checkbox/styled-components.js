// @flow
import * as React from 'react';
import styled from 'styled-components';
import { darken3 } from 'mcs-lite-theme';

export const Container: React.ComponentType<{
  size?: number,
  kind?: string,
  checked: boolean,
}> = styled.div`
  line-height: 1em;
  font-size: 12px;
  border-radius: 2px;
  width: ${props => props.size}px;
  height: ${props => props.size}px;
  box-sizing: border-box;
  user-select: none;
  cursor: pointer;
  transition: background-color 0.5s cubic-bezier(0.23, 1, 0.32, 1),
    border 0.3s cubic-bezier(0.23, 1, 0.32, 1);
  color: ${props => {
    if (props.checked) {
      return props.kind === 'default'
        ? props.theme.color.grayBase
        : props.theme.color.white;
    }
    return 'transparent';
  }};
  background-color: ${props =>
    props.checked ? props.theme.color[props.kind] : props.theme.color.white};
  border: 1px solid
    ${props =>
      props.checked
        ? darken3(props.theme.color[props.kind])
        : props.theme.color.grayDark};
`;

export const Content: React.ComponentType<{ checked: boolean }> = styled.div`
  height: 100%;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  transform: scale(${props => (props.checked ? 1 : 0)});
  transform-origin: 50% 50%;
  transition: transform 0.5s cubic-bezier(0.23, 1, 0.32, 1);

  svg {
    fill: currentColor;
  }
`;
