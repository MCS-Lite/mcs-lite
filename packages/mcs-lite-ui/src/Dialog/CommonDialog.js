import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import Dialog from "./Dialog";
import Panel from "../Panel";

const StyledPanel = styled(Panel)`
  width: 440px;
  box-shadow: 0 1px 5px 0 rgba(0, 0, 0, 0.2);

  > header {
    padding-left: 20px;
    display: flex;
    align-items: center;
  }

  > main {
    display: flex;
    flex-direction: column;
    align-items: center;

    > img {
      margin-bottom: 20px;
    }
  }

  > footer {
    display: flex;
    align-items: center;
    justify-content: center;

    > *:not(:last-child) {
      margin-right: 10px;
    }
  }
`;

const CommonDialog = ({
  component: Component,
  show,
  onHide,
  onSubmit,
  children,
  ...otherProps
}) => (
  <Dialog show={show} onHide={onHide} {...otherProps}>
    <Component onSubmit={Component === "form" ? onSubmit : null}>
      <StyledPanel>{children}</StyledPanel>
    </Component>
  </Dialog>
);

CommonDialog.displayName = "CommonDialog";
CommonDialog.propTypes = {
  show: PropTypes.bool.isRequired,
  onHide: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
  component: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
  onSubmit: PropTypes.func
};
CommonDialog.defaultProps = {
  component: "div"
};

export default CommonDialog;
