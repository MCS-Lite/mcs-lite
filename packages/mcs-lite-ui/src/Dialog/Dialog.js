import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Portal from 'react-overlays/lib/Portal';
import Transition from 'react-motion-ui-pack';
import ClickOutside from '../ClickOutside';
import emptyFunction from '../utils/emptyFunction';

export const Overlay = styled.div`
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background-color: rgba(0, 0, 0, 0.3);
  display: flex;
  align-items: center;
  flex-direction: column;
  overflow: auto;
`;

const Dialog = ({ onHide, show, children, ...otherProps }) =>
  show &&
  <Portal>
    <Transition
      component={false}
      enter={{ opacity: 1 }}
      leave={{ opacity: 0.5 }}
    >
      <Overlay key="dialog" {...otherProps}>
        <ClickOutside onClick={onHide}>
          {children}
        </ClickOutside>
      </Overlay>
    </Transition>
  </Portal>;

Dialog.propTypes = {
  children: PropTypes.node.isRequired,
  show: PropTypes.bool.isRequired,
  onHide: PropTypes.func,
};
Dialog.defaultProps = {
  onHide: emptyFunction,
};

export default Dialog;
