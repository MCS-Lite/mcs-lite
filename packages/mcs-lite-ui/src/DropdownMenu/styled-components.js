// @flow
import * as React from 'react';
import styled from 'styled-components';
import Menu from '../Menu/Menu';
import Button from '../Button';

export const StyledButton: React.ComponentType<{
  active: boolean,
}> = styled(Button)`
  display: flex;
  align-items: center;

  svg {
    margin-left: 5px;
    width: 18px;
    height: 18px;
    fill: ${props => props.theme.color.white};
    transition: transform 0.4s cubic-bezier(0.68, -0.55, 0.27, 1.55);
    transform: ${props => (props.active ? 'rotate(-180deg)' : 'initial')};
  }
`;

export const StyledMenu = styled(Menu)`
  border: 1px solid ${props => props.theme.color.grayBase};
`;
