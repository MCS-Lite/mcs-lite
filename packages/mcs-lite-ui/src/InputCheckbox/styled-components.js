// @flow
import styled from 'styled-components';

export const StyledLabel = styled.label`
  display: flex;
  align-items: center;
  cursor: pointer;
  line-height: 0;

  > *:first-child {
    margin-right: 5px;
  }
`;

export default StyledLabel;
