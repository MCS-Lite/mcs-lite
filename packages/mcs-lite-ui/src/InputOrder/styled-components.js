// @flow
import * as React from 'react';
import styled from 'styled-components';

export const Container: React.ComponentType<{
  height: number,
}> = styled.div`
  display: flex;
  flex-wrap: wrap;
  align-content: flex-start;
  border: 1px solid ${props => props.theme.color.grayBase};
  border-radius: 3px;
  height: ${props => props.height}px;
  background-color: ${props => props.theme.color.white};
  overflow: auto;

  > * {
    display: flex;
    margin: 10px;
    min-width: 45%;
    outline: none;
    cursor: pointer;
    align-items: flex-start;
  }
`;

export const Center = styled.div`
  display: flex;
  align-items: center;
`;

export const CheckboxWrapper = styled.div`
  flex-shrink: 0;
  margin-right: 5px;
`;
