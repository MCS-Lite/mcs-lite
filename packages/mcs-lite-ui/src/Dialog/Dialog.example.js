import React from 'react';
import styled from 'styled-components';
import { storiesOf } from '@storybook/react';
import Panel from '../Panel';
import Button from '../Button';
import Dialog from '.';

const StyledDialog = styled(Dialog)`
  justify-content: center;
`;

const StyledPanel = styled(Panel)`
  width: 440px;
  box-shadow: 0 1px 5px 0 rgba(0, 0, 0, 0.2);
  overflow: auto;

  > header {
    padding-left: 20px;
    display: flex;
    align-items: center;
  }

  > main {
    display: flex;
    flex-direction: column;
    align-items: center;

    > img {
      margin-bottom: 20px;
    }
  }

  > footer {
    display: flex;
    align-items: center;
    justify-content: center;

    > *:not(:last-child) {
      margin-right: 10px;
    }
  }
`;

storiesOf('Dialog', module)
  .addWithInfo(
    'API',
    '',
    () => {
      class StatefulDialog extends React.Component {
        state = { show: false };
        onShow = () => this.setState({ show: true });
        onHide = () => this.setState({ show: false });
        render() {
          return (
            <div>
              <div onClick={this.onShow}>
                Click to OPEN!
              </div>

              <Dialog show={this.state.show} onHide={this.onHide}>
                <div>Content</div>
              </Dialog>
            </div>
          );
        }
      }

      return <StatefulDialog />;
    },
    { inline: true, propTables: [Dialog] },
  )
  .addWithInfo(
    'Confirm Dialog',
    '',
    () =>
      <StyledDialog show onHide={() => {}}>
        <StyledPanel>
          <header>Notice!</header>
          <main>
            是否確定重置系統？
          </main>
          <footer>
            <Button kind="default">取消</Button>
            <Button>確定</Button>
          </footer>
        </StyledPanel>
      </StyledDialog>,
    { inline: true, propTables: [Dialog] },
  );
