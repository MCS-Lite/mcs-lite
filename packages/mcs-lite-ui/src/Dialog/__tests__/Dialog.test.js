import React from "react";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import { ThemeProvider } from "styled-components";
import { theme } from "mcs-lite-theme";
import Dialog, { Overlay } from "../Dialog";

it("should render <Dialog />", () => {
  const wrapper = mount(
    <ThemeProvider theme={theme}>
      <Dialog show>Content</Dialog>
    </ThemeProvider>
  );

  expect(toJson(wrapper.find(Dialog))).toMatchSnapshot();
});

it("should render <Overlay />", () => {
  const wrapper = mount(
    <ThemeProvider theme={theme}>
      <Overlay />
    </ThemeProvider>
  );

  expect(toJson(wrapper.find(Overlay))).toMatchSnapshot();
});
