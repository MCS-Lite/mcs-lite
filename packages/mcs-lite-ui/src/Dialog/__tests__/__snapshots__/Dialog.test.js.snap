// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <Dialog /> 1`] = `
<Dialog
  onHide={[Function]}
  show={true}
>
  <mock-portal>
    <Transition
      component={false}
      enter={
        Object {
          "opacity": 1,
        }
      }
      leave={
        Object {
          "opacity": 0.5,
        }
      }
      onEnter={[Function]}
      onLeave={[Function]}
      runOnMount={true}
    />
  </mock-portal>
</Dialog>
`;

exports[`should render <Overlay /> 1`] = `
.c0 {
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background-color: rgba(0,0,0,0.3);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: auto;
}

<Dialog__Overlay>
  <div
    className="c0"
  />
</Dialog__Overlay>
`;
